flask_app/utils.py

```diff
@@ -0,0 +1,73 @@
+import functools
+from io import StringIO
+
+import logbook
+from paramiko import AutoAddPolicy, RSAKey, SFTPClient, SSHClient
+
+logger = logbook.Logger(__name__)
+
+
+def create_key(s: str) -> str:
+    f = StringIO()
+    f.write(s)
+    f.seek(0)
+    return RSAKey.from_private_key(file_obj=f, password=None)
+
+
+class RemoteHost:
+    _TEMPDIR_COMMAND = "python -c 'import tempfile; print(tempfile.gettempdir())'"
+
+    def __init__(self, *, host, username, auth_method, pkey=None, password=None):
+        self.host = host
+        self._username = username
+        self._auth_method = auth_method
+        self._pkey = pkey
+        self._password = password
+        self._ssh_client = None
+
+    @functools.lru_cache(maxsize=None)
+    def get_os_type(self) -> str:
```

Comment:
Suggestion:
`return (self.exec_ssh_command("uname", raise_on_failure=False) or 'windows').lower()`

Reply:
What do you think about adding a utility function called "and_then" similar to rust?